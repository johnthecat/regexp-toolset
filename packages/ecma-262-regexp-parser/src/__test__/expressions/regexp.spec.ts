import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

/**
 * @see parseRegexp
 */
describe('Regexp', () => {
  describe('body', () => {
    it('should throw if there is empty input', () => {
      expect(() => parseRegexp('')).toThrowErrorMatchingInlineSnapshot(`
        "
        ❯ Can't parse input"
      `);
    });

    it('should throw if there is no starting slash', () => {
      expect(() => parseRegexp('a/')).toThrowErrorMatchingInlineSnapshot(`
        "
         ❯ a/
           ↑
         Regexp body should start with \\"/\\" symbol, like this: /.../gm"
      `);
    });

    it('should throw if there is no ending slash', () => {
      expect(() => parseRegexp('/a')).toThrowErrorMatchingInlineSnapshot(`
        "
         ❯ /a
            ↑
         Regexp body should end with \\"/\\" symbol, like this: /.../gm"
      `);
    });

    it('should throw if there is more than 1 ending slashes', () => {
      expect(() => parseRegexp('/a/[a-z]/')).toThrowErrorMatchingInlineSnapshot(`
        "
         ❯ /a/[a-z]/
              ↑
           Unexpected token"
      `);
    });
  });

  describe('flags', () => {
    it('should correctly parse flags', () => {
      expect(parseRegexp(/a/gimsuy)).toMatchSnapshot();
    });

    it('should throw if there is unsupported flag', () => {
      expect(() => parseRegexp('/a/f')).toThrowErrorMatchingInlineSnapshot(`
        "
         ❯ /a/f
              ↑
         Unknown flag 'f'"
      `);
    });

    it('should throw if there is any characters other than alphabetical', () => {
      expect(() => parseRegexp('/a/[')).toThrowErrorMatchingInlineSnapshot(`
        "
         ❯ /a/[
              ↑
         Unexpected token"
      `);
    });
  });
});
