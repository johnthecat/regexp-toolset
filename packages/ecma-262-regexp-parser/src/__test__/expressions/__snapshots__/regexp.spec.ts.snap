// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Regexp > flags > should correctly parse flags 1`] = `
{
  "body": {
    "charCode": 97,
    "end": 1,
    "kind": "Char",
    "start": 1,
    "type": "Simple",
    "value": "a",
  },
  "end": 8,
  "flags": "gimsuy",
  "kind": "Regexp",
  "start": 0,
}
`;
